packages/client/src/utils.js

```diff
@@ -106,3 +106,17 @@ export const getVaultId = (identifiers, coinType) => {
     (id) => id.indexOf(COIN_TYPE_TO_META[coinType].vaultName) >= 0
   );
 };
+
+export const showError = (modalContext, error) => {
+  modalContext.openModal(
+    <ErrorModal error={extractError(error)} />
+  );
+}
+
+function extractError(error) {
+  if (typeof error == "string" && error.includes("\n") && error.includes(": ")) {
+    const errorMessage = error.split("\n")[1];
```

Comment:
This part is hardcoded, and works for pre-conditions, post-conditions, panic and assert

Reply:
I like your idea to wrap message with special character 😄 , that seems more reliable then grabbing it from ": " as per cadence team they might restructure their error object. But if we wrap message with @@ or anything like that, that could be our unique identifier to grab the message. What do you think?

Reply:
It's not gonna work for messages that come from core contracts unfortunately, I'm not sure how to handle those, since they cannot be wrapped in special characters. What do you think how can we handle those errors?

Reply:
hmm I thought majority of the error use case that we want to surface is around our own treasury contract only. What kind of message are returning from core contracts? any examples?

Reply:
There are some examples in the PR description - when sending fungible tokens or NFT fail then the errors from those contracts can happen

Reply:
i see... in that case can we work with Sara to add a Cypress test to cover at least one UI message as part of regression suite? In case anything changed we can catch it during regression testing.

Reply:
Sounds great! And once we update the libraries in case something changes we will catch this right away and update it!

Reply:
Created a ticket here: https://linear.app/dappercollectives/issue/VSL-143/add-test-to-cover-error-scenario-in-regression-suite